src/test/java/space/obminyashka/items_exchange/controller/UserControllerIntegrationTest.java

```diff
@@ -141,36 +140,28 @@ void addChild_badTotalAmount_shouldReturnBadRequest() throws Exception {
         assertTrue(Objects.requireNonNull(mvcResult.getResolvedException()).getMessage().contains(expectedErrorMessage));
     }
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserPassword_whenOldPasswordWrong_shouldThrowInvalidDtoException() throws Exception {
+    @ParameterizedTest
+    @WithMockUser(username = "user")
+    @MethodSource("listIncorrectDataPassword")
+    void updateUserPassword_whenDataIncorrect_shouldThrowIllegalArgumentException
+            (String password, String confirmPassword, ResultMatcher status, String response)
+            throws Exception {
         when(userService.findByUsernameOrEmail(any())).thenReturn(Optional.of(user));
-        when(userService.isPasswordMatches(any(), any())).thenReturn(false);
-
-        UserChangePasswordDto userChangePasswordDto = new UserChangePasswordDto(WRONG_OLD_PASSWORD, NEW_PASSWORD, NEW_PASSWORD);
-        MvcResult mvcResult = sendDtoAndGetMvcResult(put(USER_SERVICE_CHANGE_PASSWORD), userChangePasswordDto, status().isBadRequest());
-
-        assertTrue(mvcResult.getResponse().getContentAsString().contains(getMessageSource(ResponseMessagesHandler.ValidationMessage.INCORRECT_PASSWORD)));
-    }
+        user.setPassword(CORRECT_OLD_PASSWORD);
+        MvcResult mvcResult = sendUriAndGetMvcResult(put(USER_SERVICE_CHANGE_PASSWORD).
+                param("password", password).
+                param("confirmPassword", confirmPassword), status);
 
-    @Test
-    @WithMockUser()
-    void updateUserPassword_whenUserEnteredOldPassword_shouldThrowIllegalArgumentException() throws Exception {
-        var userChangePasswordDto = new UserChangePasswordDto(CORRECT_OLD_PASSWORD, CORRECT_OLD_PASSWORD, CORRECT_OLD_PASSWORD);
-        MvcResult mvcResult = sendDtoAndGetMvcResult(put(USER_SERVICE_CHANGE_PASSWORD), userChangePasswordDto, status().isBadRequest());
         String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
 
-        assertTrue(message.contains(getMessageSource(ResponseMessagesHandler.ValidationMessage.SAME_PASSWORDS)));
+        assertTrue(message.contains(getMessageSource(response)));
     }
 
-    @Test
-    @WithMockUser(username = "admin")
-    void updateUserPassword_whenPasswordConfirmationWrong_shouldThrowIllegalArgumentException() throws Exception {
-        UserChangePasswordDto userChangePasswordDto = new UserChangePasswordDto(CORRECT_OLD_PASSWORD, NEW_PASSWORD, WRONG_NEW_PASSWORD_CONFIRMATION);
-        MvcResult mvcResult = sendDtoAndGetMvcResult(put(USER_SERVICE_CHANGE_PASSWORD), userChangePasswordDto, status().isBadRequest());
-        String message = Objects.requireNonNull(mvcResult.getResolvedException()).getMessage();
-
-        assertTrue(message.contains(getMessageSource(ResponseMessagesHandler.ValidationMessage.DIFFERENT_PASSWORDS)));
+    private static Stream<Arguments> listIncorrectDataPassword() {
```

Comment:
I don't think that's a good idea to use a parametrized test with a very single parameter. Make it a simple non-parametrized test

Reply:
refacted test